FHEM/00_SIGNALduino.pm

```diff
@@ -1348,6 +1348,14 @@ sub SIGNALduino_CheckVersionResp {
 			$hash->{logMethod}->($name, 5, "$name: CheckVersionResp, cc1101 available");
 			SIGNALduino_Get($hash, $name,"ccconf");
 			SIGNALduino_Get($hash, $name,"ccpatable");
+		} else {
+			# connect device without cc1101 to port where a device with cc1101 was previously connected (example DEF with /dev/ttyUSB0@57600) #
+			$hash->{logMethod}->($hash, 5, "$name: CheckVersionResp, delete old READINGS from cc1101 device");
+			delete($hash->{cc1101_available}) if defined( $hash->{cc1101_available} );
+
+			foreach my $value ( qw(cc1101_config cc1101_config_ext cc1101_patable) ) {
+				readingsDelete($hash,$value) if ( ReadingsVal($name, $value, undef) );
```

Comment:
Wofür wir die if Abfrage brauchen hat sich mir nicht erschlossen.
`readingsDelete($hash,$readingName) ;` sollte reichen oder?

Reply:
Ich vermeide Wege bzw. denke, eine Aktion muss ich nur ausführen wenn diese auch durchführbar ist. Aus diesem Grunde habe ich eine Abfrage drin. Darüber kann man bestimmt verschieden argumentieren.

Ich selbst denke, so Gewissheit zuhaben, erst handeln zu können wenn es auch da ist.
`"Ich kann im Laden auch nur Hartgeld ausgeben wenn ich dies habe"`

Reply:
Was willst Du mit der Abfrage erreichen? Ob das Reading existiert? Das prüfst Du doch so nicht wirklich ab und in readingsDelete wird geprüft dass nichts gelöscht wird was nicht existiert :)

Reply:
Die Abfrage sollte schauen ob aus dem Reading ein Wert kommt. Wenn das Reading nicht existent wäre, so wäre ein undef zurück gekommen und dadurch wäre auch kein Löschbefehl abgesetzt wurden.

Ich hatte in der letzten Änderung die if abfrage entfernt weil ich den Test machte mit einem Reading was nicht existierte. Da kam keine Fehlermeldung. https://github.com/RFD-FHEM/RFFHEM/pull/854#discussion_r436255686